sqlmesh/core/loader.py

```diff
@@ -307,6 +294,26 @@ def _track_file(self, path: Path) -> None:
         """Project file to track for modifications"""
         self._path_mtimes[path] = path.stat().st_mtime
 
+    def _get_variables(self, gateway_name: str) -> t.Dict[str, t.Any]:
+        if gateway_name not in self._variables_by_gateway:
+            try:
+                gateway = self.config.get_gateway(gateway_name)
+            except ConfigError:
+                from sqlmesh.core.console import get_console
+
+                get_console().log_warning(
+                    f"Gateway '{gateway_name}' not found in project '{self.config.project}'."
```

Comment:
why allow this? why not just raise?

Reply:
This logic exited before this PR, I just moved it into a helper. I'm not sure what the original motivation was.